### PR TITLE
resolves: 124 support relative filepaths

### LINKED ISSUE #124: support relative filepaths
Parse all paths of the config files to support relative filepaths. Check if the path is relative, if so, cast it to absolute from the current directory ($PWD) 

Currently, it does not work because reframe execute tests on a different directory (stage)

### PR DESCRIPTION
**Pull request by JavierCladellas**
- Closes #124 

Relative file paths for cli arguments is now supported.

For the config files, supporting relative paths for an arbitrary field is not trivial as the string might be complex (an option containing a json for example: `"--heat.json.patch='{\"op\": \"replace\",\"path\": \"/Meshes/heat/Import/filename\",\"value\": \"$cfgdir/{{parameters.meshes.value}}/case3_p{{parameters.nb_tasks.tasks.value}}.json\" }'"

I propose to keep imposing using `$PWD` in the config. Thoughts? @vincentchabannes @prudhomm 
Note that using env vars in config files is already supported. For example, `"reframe_base_dir":"$PWD/build/reframe",`

Also, I think having relative paths in config files is ambiguous. Is it relative to the file location or to the current directory ?

### CODE REVIEW

**Comment by prudhomm** [comment]
what does $PWD refer to? I would prefer a better named variable

File: config/machines/gaya.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
     "partitions":["production"],
     "prog_environment":"builtin",
     "launch_options":["-bind-to core"],
-    "reframe_base_dir":"/data/home/cladellash/benchmarking/build/reframe",
-    "reports_base_dir":"/data/home/cladellash/benchmarking/reports/",
+    "reframe_base_dir":"$PWD/build/reframe",
+    "reports_base_dir":"$PWD/reports/",
     "input_dataset_base_dir":"/data/scratch/cladellash/feelppdb/input_data",
     "output_app_dir":"/data/scratch/cladellash/feelppdb",
```
Comment (prudhomm) [inline config/machines/gaya.json:10]: What does $PWD refer to ?  it is unclear. Below we have yet other PATHs

Comment (prudhomm) [reply config/machines/gaya.json:10]: is it possible to find a better name ?

**Review by prudhomm — APPROVED** [review]

File: config/machines/gaya.json (hunk 1 above)
Comment (JavierCladellas) [reply config/machines/gaya.json:10]: $PWD is the linux environment variable containing the path of the current working directory. The name is not chosen by the application. The config files support ANY environment variable in the form `$VARIABLE`

**Pull request merged**